src/ansys/dynamicreporting/core/utils/report_remote_server.py

```diff
@@ -1033,14 +1047,118 @@ def load_templates(self, templates_json):
                 break
 
         root_attr = templates_json[root_id_str]
-        root_template = self._populate_template(root_attr, None)
+        root_template = self._populate_template(root_id_str, root_attr, None)
         self.put_objects(root_template)
         id_template_map = {}
         id_template_map[root_id_str] = root_template
         self._build_templates_from_parent(root_id_str, id_template_map, templates_json)
         self._update_changes(root_id_str, id_template_map, templates_json)
 
 
+def _check_template(template_id_str, template_attr):
+    # Check template_id_str
+    if not _check_template_name_convention(template_id_str):
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file has an invalid template name: '{template_id_str}' as the key.\n"
+            "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+        )
+
+    # Check parent and children template name convention
+    if not _check_template_name_convention(template_attr["parent"]):
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file has an invalid template name: '{template_attr['parent']}' "
+            f"that does not have the correct name convection under the key: 'parent' of '{template_id_str}'\n"
+            "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+        )
+
+    for child_name in template_attr["children"]:
+        if not _check_template_name_convention(child_name):
+            raise TemplateEditorJSONLoadingError(
+                f"The loaded JSON file has an invalid template name: '{child_name}' "
+                f"that does not have the correct name convection under the key: 'children' of '{template_id_str}'\n"
+                "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+            )
+
+    # Check key names
+    agreed_keys = {
+        "name",
+        "report_type",
+        "date",
+        "tags",
+        "params",
+        "sort_selection",
+        "item_filter",
+        "parent",
+        "children",
+    }
+    for key in template_attr.keys():
+        if key not in agreed_keys:
+            raise TemplateEditorJSONLoadingError(
+                f"The loaded JSON file is using an unknown key: '{key}' other than any in the JSON schema.\n"
+                f"Please check the '{key}' entry under '{template_id_str}' in your JSON file as you might have a typo in that entry."
+            )
+
+    # Check missing keys
+    if len(template_attr) != len(agreed_keys):  # Missing one or more entries
```

Comment:
@zhang-yuanrui  This might create issues if in the future we add more keys - the check would fail. I'd like the check to pass if we're loading a json file exported by a future version, just disregarding the new keys. For that to work, I think we should change the check to be:

len(template_attr) >= len(agreed_keys)

Reply:
According to your comments, I think the requirement is:
Skip checking any additional keys but make sure every template JSON has at least the basic several keys like "name", "report_type" and etc. If any of the basic keys are missing, throw an exception. Do I understand it correctly?